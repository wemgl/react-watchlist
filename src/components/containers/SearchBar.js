import { connect } from 'react-redux';
import { queryOMDB } from '../../actions';
import SearchBar from '../ui/SearchBar';

const mapStateToProps = state => ({
  query: state.query
});

const mapDispatchToProps = dispatch => ({
  queryOMDB(query) {
    dispatch(queryOMDB(query));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);