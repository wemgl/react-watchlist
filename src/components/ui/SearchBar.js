import React from "react"
import "../../stylesheets/searchbar.scss"
import fetch from "isomorphic-fetch"

export default class SearchBar extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            query: "",
            loading: false
        }
        this.submit = this.submit.bind(this)
    }

    submit(event) {
        event.preventDefault()
        const {query} = this.refs
        if (query.value === "") {
            return
        }

        this.setState({loading: true, query: query.value})
        fetch(`http://www.omdbapi.com/?apikey=8a16b0ce&s=${encodeURIComponent(query.value)}`)
            .then(response => {
                if (response.status >= 400) {
                    throw new Error("Bad response from server")
                }
                return response.json()
            })
            .then(json => json.Search)
            .then(movies => {
                this.setState({loading: false})
                this.props.updateMovieList(movies)
            })
            .catch(reason => {
                console.log("Error:", reason)
            })
    }

    render() {
        return (
            <form onSubmit={this.submit} className="search">
                <input className="search__bar"
                       type="text"
                       ref="query"
                       defaultValue={this.state.query}
                       name="searchbar"
                       placeholder="e.g. Kill Bill vol.2"
                />
                <button className="search__btn">Search</button>
            </form>
        )
    }
}
