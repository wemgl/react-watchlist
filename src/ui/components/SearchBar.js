import React from "react"
import "../stylesheets/search-bar.scss"

let _searchbar

const SearchBar = () =>
    <input className="searchbar"
           type="text"
           ref={ (value) => _searchbar = value }
           name="searchbar"
           placeholder="e.g. Kill Bill vol.2"
    />

export default SearchBar