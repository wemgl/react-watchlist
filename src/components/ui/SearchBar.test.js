import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBar from './SearchBar';

Enzyme.configure({ adapter: new Adapter });

const queryOMDB = (query = 'Captain America') => {
};
const query = 'Kill Bill: Vol. 2';

describe('SearchBar test', () => {
  test('receives values from props', () => {
    const component = renderer.create(
      <SearchBar
        queryOMDB={queryOMDB}
        query={query}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('input field has correct query from props', () => {
    const wrapper = mount(
      <SearchBar
        queryOMDB={queryOMDB}
        query={query}
      />
    );

    expect(wrapper.ref('query').value).toEqual('Kill Bill: Vol. 2');
  });
});