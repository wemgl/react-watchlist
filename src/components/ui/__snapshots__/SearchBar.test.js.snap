// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar test receives values from props 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <input
    className="search__bar"
    defaultValue="Kill Bill: Vol. 2"
    name="searchbar"
    placeholder="e.g. Kill Bill vol.2"
    type="text"
  />
  <button
    className="search__btn"
  >
    Search
  </button>
</form>
`;
